### PR TITLE
Adding info on the new crdb_internal tables in v21.2.

### PR DESCRIPTION
**Pull request by ianjevans**
Doc work for CRDB-2467
Doc work for CRDB-7561
Doc work for CRDB-394
Doc work for CRDB-6570

Adding information on using new `crdb_internal` tables for performance monitoring and troubleshooting.

### CODE REVIEW

File: v21.2/performance-best-practices-overview.md [hunk 1]
```diff
@@ -333,58 +333,33 @@ However, because `AS OF SYSTEM TIME` returns historical data, your reads might b
...
-Transaction contention can also increase the rate of transaction
-restarts, and thus make the proper implementation of [client-side
-transaction
-retries](transactions.html#client-side-intervention) more
-critical.
+{% include {{ page.version.version }}/performance/statement-contention.md %}
+
+### Avoid contention
 
 To avoid contention, multiple strategies can be applied:
```
Comment (stbof) [inline v21.2/performance-best-practices-overview.md:358]:
```suggestion
To avoid contention, you can apply multiple strategies:
```

File: v21.2/performance-best-practices-overview.md [hunk 2]
```diff
@@ -393,35 +368,14 @@ To avoid contention, multiple strategies can be applied:
...
-design of the schema and application. However, sometimes contention is
-unavoidable. To maximize performance in the presence of contention,
-you'll need to maximize the performance of a single range.
-To achieve this, multiple strategies can be applied:
-
-- Minimize the network distance between the replicas of a range,
-  possibly using zone configs and partitioning.
+- When replacing values in a row, use [`UPSERT`](upsert.html) and specify values for all columns in the inserted rows. This will usually have the best performance under contention, compared to combinations of [`SELECT`](select-clause.html), [`INSERT`](insert.html), and [`UPDATE`](update.html).
+
+- Increase [normalization](https://en.wikipedia.org/wiki/Database_normalization) of the data to place parts of the same records that are modified by different transactions in different tables. Increasing normalization has drawbacks as well as benefits, because denormalization can also increase performance by creating multiple copies of often-referenced data in separate ranges.
```
Comment (stbof) [inline v21.2/performance-best-practices-overview.md:373]: The second sentence is very hard to parse because it refers to both increasing normalization and denormalization.  It's not clear what the drawbacks and benefits are. 

Could you please revise it?

File: v21.2/performance-best-practices-overview.md [hunk 3]
```diff
@@ -393,35 +368,14 @@ To avoid contention, multiple strategies can be applied:
...
+
+- If the application strictly requires operating on very few different index key values, consider using [`ALTER ... SPLIT AT`](split-at.html) so that each index key value can be served by a separate group of nodes in the cluster.
+
+It is always best to avoid contention as much as possible via the design of the schema and application. However, sometimes contention is unavoidable. To maximize performance in the presence of contention, you'll need to maximize the performance of a single range. To achieve this, multiple strategies can be applied:
```
Comment (stbof) [inline v21.2/performance-best-practices-overview.md:377]:
```suggestion
It is always best to avoid contention as much as possible via the design of the schema and application. However, sometimes contention is unavoidable. To maximize performance in the presence of contention, you'll need to maximize the performance of a single range. To achieve this, you can apply multiple strategies:
```

File: v21.2/performance-best-practices-overview.md [hunk 4]
```diff
@@ -393,35 +368,14 @@ To avoid contention, multiple strategies can be applied:
...
+
+- Minimize the network distance between the replicas of a range, possibly using zone configs and partitioning.
 - Use the fastest storage devices available.
-- If the contending transactions operate on different keys within the
-  same range, add more CPU power (more cores) per node. Note however
-  that this is less likely to provide an improvement if the
-  transactions all operate on the same key.
+- If the contending transactions operate on different keys within the same range, add more CPU power (more cores) per node. Note however that this is less likely to provide an improvement if the transactions all operate on the same key.
```
Comment (stbof) [inline v21.2/performance-best-practices-overview.md:381]:
```suggestion
- If the contending transactions operate on different keys within the same range, add more CPU power (more cores) per node. However, if the transactions all operate on the same key, this may not provide an improvement .
```

**Review by stbof — CHANGES_REQUESTED** [review]
See suggestions and comment.

File: v21.1/performance-best-practices-overview.md [hunk 5]
```diff
@@ -333,58 +333,33 @@ However, because `AS OF SYSTEM TIME` returns historical data, your reads might b
...
-  multiple clients connected simultaneously to a single CockroachDB
-  cluster).
-- They operate on the same data, specifically over table rows with the
-  same index key values (either on [primary keys](primary-key.html) or
-  secondary [indexes](indexes.html)) or using index key values
-  that are close to each other, and thus place the indexed data on the
-  same [data ranges](architecture/overview.html).
+- There are multiple concurrent transactions or statements (sent by multiple clients connected simultaneously to a single CockroachDB cluster).
+- They operate on the same data, specifically over table rows with the same index key values (either on [primary keys](primary-key.html) or secondary [indexes](indexes.html)) or using index key values that are close to each other, and thus place the indexed data on the same [data ranges](architecture/overview.html).
 - At least some of the transactions write or modify the data.
```
Comment (stbof) [inline v21.1/performance-best-practices-overview.md:340]: Isn't write and modify the same thing?

File: v21.2/performance-best-practices-overview.md (hunk 2 above)
Comment (ianjevans) [reply v21.2/performance-best-practices-overview.md:373]: I'll expand this, though part of me wonders why we're even raising counter-arguments to data normalization in a section about avoiding transaction contention.

File: v21.1/performance-best-practices-overview.md [hunk 6]
```diff
@@ -393,35 +368,14 @@ To avoid contention, multiple strategies can be applied:
...
+
+- Increase [normalization](https://en.wikipedia.org/wiki/Database_normalization) of the data to place parts of the same records that are modified by different transactions in different tables. Increasing normalization has drawbacks as well as benefits, because denormalization can also increase performance by creating multiple copies of often-referenced data in separate ranges.
+
+- If the application strictly requires operating on very few different index key values, consider using [`ALTER ... SPLIT AT`](split-at.html) so that each index key value can be served by a separate group of nodes in the cluster.
+
+It is always best to avoid contention as much as possible via the design of the schema and application. However, sometimes contention is unavoidable. To maximize performance in the presence of contention, you'll need to maximize the performance of a single range. To achieve this, multiple strategies can be applied:
```
Comment (stbof) [inline v21.1/performance-best-practices-overview.md:377]:
```suggestion
It is always best to avoid contention as much as possible via the design of the schema and application. However, sometimes contention is unavoidable. To maximize performance in the presence of contention, you'll need to maximize the performance of a single range. To achieve this, you can apply multiple strategies:
```

**Review by stbof — APPROVED** [review]
